### ISSUE TITLE
Build from source is broken as ROOT submodule points to non-existent reference upstream

### ISSUE THREAD

**Issue by kgizdov** [scikit-hep/iminuit#778]
After changes were merged on the ROOT repository, source branch was deleted and the git reference was lost. See below:

![Screenshot 2022-08-06 161825](https://user-images.githubusercontent.com/3164953/183250635-9cc22d12-365b-423f-8368-a71f0593888f.png)

Error log:
```bash
Submodule path 'extern/pybind11': checked out '914c06fb252b6cc3727d0eedab6736e88a3fcb01'
fatal: git upload-pack: not our ref a4bb8f3342bcbf4d1e91822c8fdb05232f7c34d0
fatal: remote error: upload-pack: not our ref a4bb8f3342bcbf4d1e91822c8fdb05232f7c34d0
fatal: Fetched in submodule path 'extern/root', but it did not contain a4bb8f3342bcbf4d1e91822c8fdb05232f7c34d0. Direct fetching of that commit failed.
```

Please update the `extern/submodule` to point to the correct reference upstream.

**Comment by HDembinski** [comment]
Please give exact steps to reproduce. I don't know what you are talking about.

**Comment by HDembinski** [comment]
Also, why are you building from source?

**Comment by HDembinski** [comment]
`git clone --recursive git@github.com:scikit-hep/iminuit.git` works for me

**Comment by kgizdov** [comment]
Right, well, maybe let me answer your questions next time.

Firstly, I am building from source as I'm maintaining a lot of CERN/HEP software for Arch Linux, and it's policy to build everything in a clean chroot from source to make sure packages are reproducible.

Secondly, `git clone --recursive` only works because of a GitHub's quirk specifically designed to make life easy. As in GitHub has kept an already deleted branch just in case. However, the reference this repo points (`a4bb8f3342bcbf4d1e91822c8fdb05232f7c34d0`) to no longer exists at ROOT's repository. You can verify this in two ways. One is:
```bash
$ git clone https://github.com/root-project/root
Cloning into 'root'...
remote: Enumerating objects: 818935, done.
remote: Counting objects: 100% (858/858), done.
remote: Compressing objects: 100% (502/502), done.
remote: Total 818935 (delta 399), reused 743 (delta 348), pack-reused 818077
Receiving objects: 100% (818935/818935), 944.38 MiB | 27.89 MiB/s, done.
Resolving deltas: 100% (582182/582182), done.
Updating files: 100% (21507/21507), done.
$ cd root
$ git checkout a4bb8f3342bcbf4d1e91822c8fdb05232f7c34d0
fatal: reference is not a tree: a4bb8f3342bcbf4d1e91822c8fdb05232f7c34d0
```
Or, with a browser, open the GitHub page on your repo, go to `extern/root`, and you'll be greeted with the screenshot I posted above.

In conclusion, when I try to build this package in a reproducible Git-consistent way from source, which circumvents GitHub's specific quirks, it fails to find the reference as it's not there. This is the case because your original branch was deleted upstream as part of the merged Pull Request.

So you can either do nothing about it, which will make me carry a specific workaround, or comply with Git by updating the reference to the correct upstream one `195f87330295e1ecdace9e99fc1b7e9558bf61d4`. Cheers and thanks,

**Comment by HDembinski** [comment]
Thanks, now this is a report that I can work with. Let me see how I can fix this.

**Comment by HDembinski** [comment]
> Firstly, I am building from source as I'm maintaining a lot of CERN/HEP software for Arch Linux, and it's policy to build everything in a clean chroot from source to make sure packages are reproducible.

May I suggest then that you use the Python source distribution and generate your package from that, not from the source on Github? You can download the source distribution from PyPI, e.g. https://files.pythonhosted.org/packages/f3/90/0c22673a8e0f10be20b2337a2ad35ff428818b1d54b5f7dbc300f4f5c577/iminuit-2.15.2.tar.gz
This is what a source distribution is for and it isolates you from the quirks of GitHub and me referring to CERN's ROOT with my patches applied.

**Comment by HDembinski** [comment]
It is fixed now.
